tests/ui/no_std/simple-runs.rs

```diff
@@ -0,0 +1,33 @@
+//! Check that `no_std` binaries can link and run without depending on `libstd`.
+
+//@ run-pass
+//@ compile-flags: -Cpanic=abort
+//@ ignore-wasm different `main` convention
```

Comment:
Unsure if we need to add more to the ignores here? Is something like `ignore-embedded` a thing, and would it make sense?

Reply:
I'm guessing `run-pass` already omits platforms that can't run standalone executables, cc @jieyouxu for a better answer.

Reply:
See https://github.com/rust-lang/rust/blob/4fd8c04da0674af2c51310c9982370bfadfa1b98/src/tools/compiletest/src/runtest.rs#L986, I believe in the general case `run-pass` intentionally does not try to be "smart" and omit platforms that can't run standalone executables. Generally tests will need `ignore-cross-compile` if they need to run the test binary (but that only runs on host).